inc/console/migration/domainsplugintocorecommand.class.php

```diff
@@ -0,0 +1,745 @@
+               'comment'               => $dom['comment'],
+               'others'                => $dom['others'],
+               'is_helpdesk_visible'   => $dom['is_helpdesk_visible'],
+               'date_mod'              => $dom['date_mod'],
+               'is_deleted'            => $dom['is_deleted']
+            ];
+            $domain_input = Toolbox::addslashes_deep($domain_input);
+
+            $domain = new Domain();
+            if ($core_dom !== null) {
+               $res = (bool)$domain->update($domain_input + ['id' => $core_dom]);
+            } else {
+               $new_did = (int)$domain->add($domain_input);
+               $res = $new_did > 0;
+               if ($res) {
+                  $core_domains[$dom['name']] = $new_did;
+               }
+            }
+
+            if (!$res) {
+               $has_errors = true;
+
+               $message = sprintf(
+                  $core_dom === null ?
+                     __('Unable to add domain %s.') :
+                     __('Unable to update domain %s.'),
+                  $dom['name']
+               );
+               $this->outputImportError($message, $progress_bar);
+               return false;
+            }
+
+            $this->addElementToMapping(
+               'PluginDomainsDomains',
+               $dom['id'],
+               'Domain',
+               $new_did ?? $core_dom
+            );
+         }
+
+         $progress_bar->finish();
+         $this->output->write(PHP_EOL);
+      } else {
+         $this->output->writeln(
+            '<comment>' . __('No domains found.') . '</comment>',
+            OutputInterface::VERBOSITY_NORMAL
+         );
+      }
+
+      return !$has_errors;
+   }
+
+   /**
+    * Migrate domain items
+    *
+    * @return boolean
+    */
+   protected function importDomainItems() {
+      $has_errors = false;
+
+      $this->output->writeln(
+         '<comment>' . __('Importing domains items...') . '</comment>',
+         OutputInterface::VERBOSITY_NORMAL
+      );
+
+      $items_iterator = $this->db->request([
+         'FROM'   => 'glpi_plugin_domains_domains_items',
+         'ORDER'  => 'id ASC'
+      ]);
+
+      $core_items = [];
+      $coreitems_iterator = $this->db->request([
+         'FROM'   => Domain_Item::getTable()
+      ]);
+      while ($row = $coreitems_iterator->next()) {
+         $core_items[$row['domains_id'].$row['itemtype'].$row['items_id']] = $row['id'];
+      }
+
+      if ($items_iterator->count()) {
+         $progress_bar = new ProgressBar($this->output, $items_iterator->count());
+         $progress_bar->start();
+
+         foreach ($items_iterator as $itm) {
+            $progress_bar->advance(1);
+            $core_item = null;
+            $domains_id = $this->getCorrespondingItem('PluginDomainsDomains', $itm['plugin_domains_domains_id'])->fields['id'];
+
+            if (isset($core_items[$domains_id.$itm['itemtype'].$itm['items_id']])) {
```

Comment:
If relation already exists in core, as you do not update anything except setting `domainrelations_id` to 0, the update part seems useless. I guess just outputing a skipping message and continue to next element will be enough.